### ISSUE TITLE
Panic when using read_box

### ISSUE THREAD

**Issue by frewsxcv** [mozilla/mp4parse-rust#4]
```rust
extern crate mp4parse;

use std::io::Cursor;

fn main() {
    let mut c = Cursor::new(b"\x00\x00\x00\x01\x30\x30\x30\x30\x00\x00\x00\x00\x00\x00\x00\x00".to_vec());
    let mut context = mp4parse::MediaContext::new();
    let _ = mp4parse::read_box(&mut c, &mut context);
}
```

```
coreyf@frewbook-pro /t/mp4 (master)> cargo run
     Running `target/debug/mp4`
thread '<main>' panicked at 'assertion failed: size64 >= 16', /private/tmp/mp4parse-rust/src/lib.rs:273
Process didn't exit successfully: `target/debug/mp4` (exit code: 101)
```

panic discovered using [afl.rs](https://github.com/frewsxcv/afl.rs)

**Comment by kinetiknz** [comment]
Thanks for the report.  I've converted this into error return in https://github.com/kinetiknz/mp4parse-rust/commit/a60ea2d6f49c91a763a0dbe331259f8deba2f460, Ralph will probably merge it tomorrow.

Since this project is very early in development we've been using panics and asserts as placeholders for areas where more thought about the code is needed.  Since that's going to make fuzzing difficult, I've converted the remaining ones into error returns in my branch... so it's probably worth testing against my branch at https://github.com/kinetiknz/mp4parse-rust/tree/experimental for now, at least until Ralph merges the changes upstream.

**Comment by rillian** [comment]
Thanks for that, kinetik. I've merged the changes and added this testcase in abb0f2b5c43d94c32aa26a6f8936216584517707, confirming the fix. Hopefully the next one you find will be a real bug!
